File: Serene/Serene.Web/Modules/Common/Helpers/BasicProgressDialog.ts
namespace Serene {

    export class BasicProgressDialog extends Serenity.TemplatedDialog<any> {

        private cancelled: boolean;
        private cancelTitle: string;

        constructor() {
            super();

            this.byId('ProgressBar').progressbar({
                max: 100,
                value: 0,
                change: (e, v) => {
                    this.byId('ProgressLabel').text(this.get_value() + ' / ' + this.get_max());
                }
            });
        }

        public get_cancelled() {
            return this.cancelled;
        }

        public set_cancelled(value) {
            this.cancelled = value;
        }

        public get_max() {
            return this.byId('ProgressBar').progressbar().progressbar('option', 'max');
        }

        public set_max(value) {
            this.byId('ProgressBar').progressbar().progressbar('option', 'max', value);
        }

        public get_value() {
            return this.byId('ProgressBar').progressbar('value');
        }

        public set_value(value) {
            this.byId('ProgressBar').progressbar().progressbar('value', value);
        }

        public get_title() {
            return this.element.dialog().dialog('option', 'title');
        }

        public set_title(value) {
            this.element.dialog().dialog('option', 'title', value);
        }

        public get_cancelTitle() {
            return this.cancelTitle;
        }

        public set_cancelTitle(value) {
            this.cancelTitle = value;
        }

        getDialogOptions() {
            var opt = super.getDialogOptions();
            opt.title = Q.text('Site.BasicProgressDialog.PleaseWait');
            opt.width = 600;
            opt.buttons = [{
                text: Q.text('Dialogs.CancelButton'),
                click: () => {
                    this.set_cancelled(true);
                    this.element.closest('.ui-dialog')
                        .find('.ui-dialog-buttonpane .ui-button')
                        .attr('disabled', 'disabled')
                        .css('opacity', '0.5');

                    this.element.dialog('option', 'title', Q.trimToNull(this.get_cancelTitle()) ||
                        Q.text('Site.BasicProgressDialog.CancelTitle'));
                }
            }];

            return opt;
        }

        initDialog() {
            super.initDialog();
            this.element.closest('.ui-dialog').find('.ui-dialog-titlebar-close').hide();
        }

        getTemplate() {
            return (
                "<div class='s-DialogContent s-BasicProgressDialogContent'>" +
                    "<div id='~_StatusText' class='status-text' ></div>" +
                    "<div id='~_ProgressBar' class='progress-bar'>" +
                        "<div id='~_ProgressLabel' class='progress-label' ></div>" +
                    "</div>" +
                "</div>");
        }
    }
}

File: Serene/Serene.Script/Common/Helper/BasicProgressDialog.cs
/* 
This class has been ported to TypeScript. See BasicProgressDialog.ts
Code below is only a reference for those who want to use Saltaralle

namespace Serene
{
    using jQueryApi.UI.Widgets;
    using Serenity;
    using System.Collections.Generic;

    public class BasicProgressDialog : TemplatedDialog<object>
    {
        public BasicProgressDialog()
        {
            var self = this;
            this.ById("ProgressBar").ProgressBar(new ProgressBarOptions
            {
                Max = 100,
                Value = 0,
                OnChange = (e, v) =>
                {
                    self.ById("ProgressLabel").Text(self.Value + " / " + self.Max);
                }
            });
        }

        public bool Cancelled { get; set; }

        public int Max
        {
            get { return this.ById("ProgressBar").ProgressBar().Max; }
            set { this.ById("ProgressBar").ProgressBar().Max = value; }
        }

        public int Value
        {
            get { return (int)this.ById("ProgressBar").As<dynamic>().progressbar("value"); }
            set { this.ById("ProgressBar").ProgressBar().Value(value); }
        }

        public string Title
        {
            get { return this.Element.Dialog().Title; }
            set { this.Element.Dialog().Title = value; }
        }

        public string CancelTitle { get; set; }

        protected override DialogOptions GetDialogOptions()
        {
            var self = this;
            var opt = base.GetDialogOptions();
            opt.Title = Q.Text("Site.BasicProgressDialog.PleaseWait");
            opt.Width = 600;
            opt.Buttons = new List<DialogButton>
            {
                new DialogButton {
                    Text = Q.Text("Dialogs.CancelButton"),
                    Click = delegate {
                        self.Cancelled = true;

                        self.Element.Closest(".ui-dialog")
                            .Find(".ui-dialog-buttonpane .ui-button")
                            .Attribute("disabled", "disabled")
                            .CSS("opacity", "0.5");

                        self.Element.Dialog().Title = CancelTitle.TrimToNull() ?? 
                            Q.Text("Site.BasicProgressDialog.CancelTitle");
                    }
                }
            };

            return opt;
        }

        protected override void InitDialog()
        {
            base.InitDialog();

            this.element.Closest(".ui-dialog").Find(".ui-dialog-titlebar-close").Hide();
        }

        protected override string GetTemplate()
        {
            return
                "<div class='s-DialogContent s-BasicProgressDialogContent'>" +
                    "<div id='~_StatusText' class='status-text'></div>" +
                    "<div id='~_ProgressBar' class='progress-bar'>" +
                        "<div id='~_ProgressLabel' class='progress-label'></div>" +
                    "</div>" +
                "</div>";
        }
    }
}
*/